app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -422,7 +422,7 @@ def repos_in_env_cv(environment = nil, content_view = nil)
         repos = Katello::Repository
         repos = repos.in_environment(environment) if environment
         repos = repos.in_content_views([content_view]) if content_view
-        repos
+        repos.class == Class ? [] : repos
```

Comment:
It'd be nice if this code returned an active record relation albeit an empty one in this case. side note: this method is really weird because it seems to implicitly require cv and/or env params which your change uncovered. I don't wanna tackle 'improving' this method here though.

You could have it return an empty relation like this: `repos.respond_to?(:to_a) ? repos : repos.none` I thought not checking the class may be *slightly* better because we don't actually are about the class, but if it is array-like. Thoughts?

Reply:
works for me!